From patchwork Fri Jun  9 06:57:57 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: drm: hibmc: Use set_busid function from drm core
From: Daniel Axtens <dja@axtens.net>
X-Patchwork-Id: 9780141
Message-Id: <20170609065757.9736-1-dja@axtens.net>
To: dri-devel@lists.freedesktop.org
Cc: Xinliang Liu <z.liuxinliang@hisilicon.com>,
 Rongrong Zou <zourongrong@gmail.com>, Daniel Axtens <dja@axtens.net>
Date: Fri,  9 Jun 2017 16:57:57 +1000
Patch-mainline: Never, fixed with code rework: https://patchwork.kernel.org/patch/9780141
References: bsc#1067977

Currently, calling drmGetBusid from libdrm on a hibmc VGA
card returns a string like "0007:a1:00.0".

The busid reported by most cards begins with "pci:". For
example, on an amd64 system, a VGA card reported
"pci:0000:00:02.0".

The missing "pci:" prefix confuses Xorg and leads to crashes
and other misbehaviour.[0]

Use the standard helper from the drm core to set the busid to
include the "pci:" prefix. This is done by many other drivers.

With this patch, Xorg can be run successfully.

[0]: https://bugs.launchpad.net/ubuntu/+source/xorg/+bug/1691991

Cc: Xinliang Liu <z.liuxinliang@hisilicon.com>
Cc: Rongrong Zou <zourongrong@gmail.com>
Signed-off-by: Daniel Axtens <dja@axtens.net>
Signed-off-by: Matthias Brugger <mbrugger@suse.com> 
Signed-off-by: Michal Srb <msrb@suse.com>
---
 drivers/gpu/drm/hisilicon/hibmc/hibmc_drm_drv.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/hisilicon/hibmc/hibmc_drm_drv.c b/drivers/gpu/drm/hisilicon/hibmc/hibmc_drm_drv.c
index 2ffdbf9801bd..9663a633897a 100644
--- a/drivers/gpu/drm/hisilicon/hibmc/hibmc_drm_drv.c
+++ b/drivers/gpu/drm/hisilicon/hibmc/hibmc_drm_drv.c
@@ -69,6 +69,7 @@ static struct drm_driver hibmc_driver = {
 	.dumb_map_offset        = hibmc_dumb_mmap_offset,
 	.dumb_destroy           = drm_gem_dumb_destroy,
 	.irq_handler		= hibmc_drm_interrupt,
+	.set_busid		= drm_pci_set_busid,
 };
 
 static int __maybe_unused hibmc_pm_suspend(struct device *dev)
